src/Fabulous.Avalonia/Widgets/Controls/NumericUpDown.fs

```diff
@@ -0,0 +1,176 @@
+﻿namespace Fabulous.Avalonia
+
+open System.Runtime.CompilerServices
+open Avalonia.Controls
+open Fabulous
+open System
+open Avalonia.Interactivity
+
+type IFabNumericUpDown =
+    inherit IFabTemplatedControl
+
+module NumericUpDown =
+    let WidgetKey = Widgets.register<NumericUpDown> ()
+
+    let AllowSpin =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.AllowSpinProperty
+
+    let ButtonSpinnerLocation =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.ButtonSpinnerLocationProperty
+
+    let ClipValueToMinMax =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.ClipValueToMinMaxProperty
+
+    let FormatString =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.FormatStringProperty
+
+    let HorizontalContentAlignment =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.HorizontalContentAlignmentProperty
+
+    let VerticalContentAlignment =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.VerticalContentAlignmentProperty
+
+    let Increment =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.IncrementProperty
+
+    let IsReadOnly =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.IsReadOnlyProperty
+
+    let Maximum =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.MaximumProperty
+
+    let Minimum =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.MinimumProperty
+
+    let NumberFormat =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.NumberFormatProperty
+
+    let ParsingNumberStyle =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.ParsingNumberStyleProperty
+
+    let ShowButtonSpinner =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.ShowButtonSpinnerProperty
+
+    let Text = Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.TextProperty
+
+    let TextConverter =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.TextConverterProperty
+
+    let Watermark =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.WatermarkProperty
+
+    let Value =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.ValueProperty
+
+    let ValueChanged =
+        Attributes.defineAvaloniaPropertyWithChangedEvent' "NumericUpDown_ValueChanged" NumericUpDown.ValueProperty
+
+    let LostFocus =
+        Attributes.defineEvent<RoutedEventArgs> "NumericUpDown_LostFocus" (fun target ->
+            (target :?> NumericUpDown).LostFocus)
+
+[<AutoOpen>]
+module NumericUpDownBuilders =
+    type Fabulous.Avalonia.View with
+
+        static member inline NumericUpDown<'msg>(value: Nullable<decimal>, valueChanged: Nullable<decimal> -> 'msg) =
```

Comment:
See my previous comments. This now can we a `decimal option` and be more F# friendly :) 

```fsharp
static member inline NumericUpDown<'msg>(value: decimal option, valueChanged: decimal option -> 'msg) =
            WidgetBuilder<'msg, IFabNumericUpDown>(
                NumericUpDown.WidgetKey,
                NumericUpDown.Value.WithValue(value),
                NumericUpDown.ValueChanged.WithValue(ValueEventData.create value (fun args -> valueChanged args |> box))
            )
```